python-client/giskard/models/huggingface/__init__.py

```diff
@@ -0,0 +1,111 @@
+import uuid
+from typing import Union
+import logging
+
+from scipy import special
+import torch
+from pathlib import Path
+import yaml
+
+from giskard.core.core import SupportedModelTypes
+from giskard.core.model import WrapperModel
+
+from transformers import pipeline, pipelines
+
+
+logger = logging.getLogger(__name__)
+
+
+class HuggingFaceModel(WrapperModel):
+
+    def __init__(self,
+                 clf,
+                 model_type: Union[SupportedModelTypes, str],
+                 name: str = None,
+                 data_preprocessing_function=None,
+                 model_postprocessing_function=None,
+                 feature_names=None,
+                 classification_threshold=0.5,
+                 classification_labels=None) -> None:
+
+        super().__init__(clf=clf,
+                         model_type=model_type,
+                         name=name,
+                         data_preprocessing_function=data_preprocessing_function,
+                         model_postprocessing_function=model_postprocessing_function,
+                         feature_names=feature_names,
+                         classification_threshold=classification_threshold,
+                         classification_labels=classification_labels)
+
+        self.huggingface_module = clf.__class__
+        self.pipeline_task = clf.task if isinstance(clf, pipelines.Pipeline) else None
+
+    # This works, but the problem with AutoModels (loaded model in that context) is that their
+    # output are different from the original saved model output
+    """@classmethod
+    def load_clf(cls, local_path):
+        huggingface_meta_file = Path(local_path) / 'giskard-model-huggingface-meta.yaml'
+        if huggingface_meta_file.exists():
+            with open(huggingface_meta_file) as f:
+                huggingface_meta_file = yaml.load(f, Loader=yaml.Loader)
+
+        return AutoModel.from_pretrained(local_path, from_tf=huggingface_meta_file["from_tf"])"""
+
+    @classmethod
+    def load_clf(cls, local_path):
+        huggingface_meta_file = Path(local_path) / 'giskard-model-huggingface-meta.yaml'
+        if huggingface_meta_file.exists():
+            with open(huggingface_meta_file) as f:
+                huggingface_meta_file = yaml.load(f, Loader=yaml.Loader)
```

Comment:
It's going to be a little cleaner if you create a new variable `huggingface_meta` to store the loaded meta, not reuse the Path variable